src/gateway/WebSocketClient.ts

```diff
@@ -0,0 +1,127 @@
+import WebSocket from 'ws';
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import { EVENTS, Payload } from './constants';
+import { GATEWAY_OPCODES, GATEWAY_CLOSE_EVENT_CODES } from '../util/Constants';
+
+export default class WebSocketClient {
+  private _intents: number;
+  private _token: string;
+  private _wsURL: string;
+  private heartBeatInterval!: NodeJS.Timeout;
+  private lastHeartbeatAck = true;
+  private seq!: number;
+  private session_id!: string;
+  private ws!: WebSocket;
+
+  constructor(token: string, wsURL: string, intents: number) {
+    this._token = token;
+    this._wsURL = wsURL;
+    this._intents = intents;
+
+    this.identify = this.identify.bind(this);
+    this.onClose = this.onClose.bind(this);
+    this.onMessage = this.onMessage.bind(this);
+  }
+
+  initialize() {
+    this.ws = new WebSocket(this._wsURL);
+    this.ws.on('open', this.identify);
+    this.ws.on('close', this.onClose);
+    this.ws.on('message', this.onMessage);
+  }
+
+  private heartbeat(expected: boolean) {
```

Comment:
We should prefix these with `_` even though they're marked private

Reply:
Original reviews https://github.com/libraryofcode/dis.ts/pull/15#discussion_r548977896 https://github.com/libraryofcode/dis.ts/pull/15#discussion_r548978018

Reply:
We have `initialize` for that but I don't know if that's a great name, also `initialize` will currently connect to the gateway (ref https://github.com/libraryofcode/dis.ts/pull/15#discussion_r548978209)

Reply:
> 
> 
> We should prefix these with `_` even though they're marked private

You want to prefix all private functions with `_`?

Reply:
> 
> 
> We have `initialize` for that but I don't know if that's a great name, also `initialize` will currently connect to the gateway (ref [#15 (comment)](https://github.com/libraryofcode/dis.ts/pull/15#discussion_r548978209))

yeah it will, what name do you suggest?

Reply:
> You want to prefix all private functions with `_`?

yes please

Reply:
> yeah it will, what name do you suggest?
`connect` or `start` as Khaaz mentioned

Reply:
alright both done, we'll use connect()